en/developer/frameworks/articles/configuration/08-customizing-the-ui/03-configuration-forms.markdown

```diff
@@ -0,0 +1,227 @@
+	paginationMode = com.liferay.dynamic.data.mapping.model.DDMFormLayout.SINGLE_PAGE_MODE,
+	value = {
+		@DDMFormLayoutPage(
+			{
+				@DDMFormLayoutRow(
+					{
+						@DDMFormLayoutColumn(
+							size = 12,
+							value = {
+								"imageCheckToken", "imageDefaultUseInitials",
+								"imageMaxSize"
+							}
+						)
+					}
+				),
+				@DDMFormLayoutRow(
+					{
+						@DDMFormLayoutColumn(
+							size = 6, value = "imageMaxHeight"
+						),
+						@DDMFormLayoutColumn(size = 6, value = "imageMaxWidth")
+					}
+				)
+			}
+		)
+	}
+)
+public interface MyFooConfigurationForm {
+```
+
+![Figure 1: The DDM annotations are used to lay out this configuration form.](../../../images/configuration-ddm-form.png)
+
+### Writing the Form Declaration
+
+Create a new implementation of `ConfigurationDDMFormDeclaration` to register
+your new configuration form class:
+```java
+package com.liferay.docs.my.foo.configuration.definition;
+
+import com.liferay.configuration.admin.definition.ConfigurationDDMFormDeclaration;
+import org.osgi.service.component.annotations.Component;
+...
+
+@Component(
+	immediate = true,
+	property = "configurationPid=com.liferay.docs.my.foo.configuration.MyFooConfiguration",
+	service = ConfigurationDDMFormDeclaration.class
+)
+public class MyFooConfigurationDDMFormDeclaration
+	implements ConfigurationDDMFormDeclaration {
+
+	@Override
+	public Class<?> getDDMFormClass() {
+		return MyFooConfigurationForm.class;
+	}
+
+}
+```
+
+Now your configuration class is backed by the form-building power of Liferay's
+native [Forms application](/docs/7-2/user/-/knowledge_base/u/forms).
+
+To see how this is done for one of Liferay's own configurations, check out all
+of the configuration classes for the User Images configuration (Control Panel
+&rarr; Configuration &rarr; System Settings &rarr; User Images):
+
+[`UserFileUploadsConfigurationForm`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-api/src/main/java/com/liferay/users/admin/configuration/definition/UserFileUploadsConfigurationForm.java)
+
+[`UserFileUploadsConfiguration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-api/src/main/java/com/liferay/users/admin/configuration/UserFileUploadsConfiguration.java)
+
+[`UserFileUploadsConfigurationBeanDeclaration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/configuration/definition/UserFileUploadsConfigurationBeanDeclaration.java)
+
+[`UserFileUploadsConfigurationDDMFormDeclaration.java`](https://github.com/liferay/liferay-portal/blob/7.2.0-ga1/modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/configuration/definition/UserFileUploadsConfigurationDDMFormDeclaration.java)
+
+## Additional Details
+
+Note: we don't have docs for the Forms DDM annotations (I know because I write
```

Comment:
Well, that's what we included in 7.1, but we should have more, and we should have it for 7.2. But we don't need it here, since that's not what we're showing.